core/src/org/pentaho/di/core/Const.java

```diff
@@ -2420,7 +2444,13 @@ public static int indexOfString( String lookup, List<String> list ) {
             addSplit = oddNumberOfEnclosures;
           }
           if ( addSplit ) {
-            splitList.add( concatSplit.toString() );
+            String splitResult = concatSplit.toString();
+            //remove enclosure from resulting split
+            if ( removeEnclosure ) {
+              splitResult = splitResult.replaceAll( enclosure, "" );
```

Comment:
what about "ab""c", "de" ? What's the result there ?

Because I think doubling the quote is the way to escape quotes in CSV (can you please confirm that). The expected result there would be ["ab"c"]["de"] before fix and [ab"c][de] after fix.